### PR TITLE
Group follow-up emails by recipient to prevent duplicate sends

### PR DESCRIPTION
**Pull request by holdenk**
## Summary
Implement email grouping logic in the follow-up email sender to ensure each recipient receives at most one follow-up email per batch, even if multiple denials are pending for the same email address. When multiple follow-ups are due for the same recipient, the most significant one (longest duration: 90-day > 30-day > 7-day) is sent, and the others are marked as sent without actually emailing.

## Key Changes

- **Email grouping in `FollowUpEmailSender`**:
  - Added `_group_candidates_by_email()` method that groups pending follow-ups by email address and selects the best candidate (longest duration) per email
  - Added `_mark_as_sent_without_sending()` helper to mark suppressed follow-ups as sent in the database without sending emails

- **Updated `send_all()` method**:
  - Now groups candidates by email before processing
  - Sends only the best candidate per email address
  - Marks other grouped candidates as sent without emailing
  - Logs suppressed follow-ups for visibility

- **Added async support**:
  - Implemented `asend_all()` method with the same grouping logic
  - Includes rate limiting with random delays between sends
  - Uses `sync_to_async` for database operations

- **Management command improvements**:
  - Updated `send_followup_emails` command to display both total pending records and unique email count
  - Dry-run mode now shows suppressed follow-up counts per recipient
  - Uses new `find_candidates()` method instead of `_find_candidates()`

## Implementation Details

- Grouping uses `defaultdict` to organize candidates by email address
- Best candidate selection sorts by `follow_up_type.duration` in descending order
- Suppressed follow-ups are marked with current timestamp to maintain audit trail
- Both sync and async paths support the `count` parameter to limit unique emails processed
- Comprehensive test coverage including edge cases (failures, different emails, duration selection)

https://claude.ai/code/session_01QsSDRMusP22NR7F9eqhptd

## Release Notes

* **New Features**
  * Follow-up emails are now sent in intelligent batches—one email per unique email address, with the system automatically selecting the longest-duration follow-up type when multiple follow-ups are pending for the same recipient.

* **Improvements**
  * Status reporting in follow-up email commands now displays unique recipient counts and suppression counts for better visibility into pending emails.

* **Tests**
  * Added comprehensive test coverage for follow-up email grouping and batch processing logic.

### CODE REVIEW

File: fighthealthinsurance/followup_emails.py [hunk 1]
```diff
@@ -217,5 +269,33 @@ def dosend(
...
+            candidates = await self.afind_candidates()
+        grouped = self._group_candidates_by_email(candidates)
+        if count is not None:
+            grouped = grouped[:count]
+        sent = 0
+        for best, others in grouped:
+            result = await self.adosend(follow_up_sched=best)
+            if result:
+                if others:
+                    await sync_to_async(self._mark_as_sent_without_sending)(others)
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/followup_emails.py]: Good catch — this was a real bug. Fixed in e622efb.

The grouped send now iterates candidates in priority order: stale ones are detected via a new `_is_stale()` check and skipped (marked sent without emailing). The first non-stale candidate gets actually sent via `dosend()`, and only then are the remaining candidates suppressed. If all candidates in the group are stale, they all get marked sent with no email (correct behavior).

Added a test (`test_send_all_skips_stale_best_sends_next`) covering the exact scenario: denial A's 7-day is stale (30-day already sent), denial B's 7-day is valid — verifies that denial B's follow-up actually sends.

---
_Generated by [Claude Code](https://claude.ai/code)_
